### PR TITLE
Expose API to set non earth DEMs flag

### PR DESCRIPTION
**Pull request by adityapande-1995**
🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

# 🎉 New feature

## Summary
This PR exposes the ``isNonEarthDEM`` member so that it can be set before the ``Load`` function, if required.
Needed for this PR : https://github.com/gazebosim/gz-sim/pull/1478

## Test it
Added some lines to the existing Moon DEM load test case.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: geospatial/include/gz/common/geospatial/Dem.hh [hunk 1]
```diff
@@ -49,6 +49,11 @@ namespace ignition
       /// \return 0 when the operation succeeds to open a file.
       public: int Load(const std::string &_filename = "");
 
+      /// \brief Indicate that this is a non Earth DEM.
+      /// \param[in] isNonEarthDem Should be true if this is a
+      /// non earth DEM, otherwise false.
+      public: void SetNonEarthDEM(bool isNonEarthDem);
```
Comment (ahcorde) [inline geospatial/include/gz/common/geospatial/Dem.hh:55]:
```suggestion
      public: void SetNonEarthDEM(bool _isNonEarthDem);
```

File: geospatial/src/Dem.cc [hunk 2]
```diff
@@ -82,6 +82,12 @@ Dem::~Dem()
     GDALClose(reinterpret_cast<GDALDataset *>(this->dataPtr->dataSet));
 }
 
+//////////////////////////////////////////////////
+void Dem::SetNonEarthDEM(bool isNonEarthDem)
```
Comment (ahcorde) [inline geospatial/src/Dem.cc:86]:
```suggestion
void Dem::SetNonEarthDEM(bool _isNonEarthDem)
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
test?

File: geospatial/include/gz/common/geospatial/Dem.hh (hunk 1 above)
Comment (adityapande-1995) [reply geospatial/include/gz/common/geospatial/Dem.hh]: Fixed

File: geospatial/src/Dem.cc (hunk 2 above)
Comment (adityapande-1995) [reply geospatial/src/Dem.cc]: Fixed

**Review by WilliamLewww — APPROVED** [review]

File: geospatial/include/gz/common/geospatial/Dem.hh [hunk 3]
```diff
@@ -49,6 +49,16 @@ namespace ignition
       /// \return 0 when the operation succeeds to open a file.
       public: int Load(const std::string &_filename = "");
 
+      /// \brief Indicate that this is a non Earth DEM.
+      /// \param[in] isNonEarthDem Should be true if this is a
```
Comment (ahcorde) [inline geospatial/include/gz/common/geospatial/Dem.hh:53]:
```suggestion
      /// \param[in] _isNonEarthDem Should be true if this is a
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by mjcarroll** [comment]
Was there a decision to revisit this variable naming?

File: geospatial/include/gz/common/geospatial/Dem.hh (hunk 3 above)
Comment (adityapande-1995) [reply geospatial/include/gz/common/geospatial/Dem.hh]: Resolved : https://github.com/gazebosim/gz-common/pull/353/commits/8bcc9d0ea068aa8992a9e0061ceedd43af7e95fd

**Review by ahcorde — APPROVED** [review]

**Pull request merged**